### PR TITLE
Add test coverage

### PR DESCRIPTION
**Pull request by grahamhar**
Add test coverage.  …
I think I have found two issues in providing these tests which has led
to them being marked to be skipped.

TODO: tests for FixedJSONFileCache class still needed

Warnings are from external dependancies so can be ignored but I've
chosen not to silence them to act as a reminder to upgrade.

Coverage report added (Vanity rather than to show good test coverage)

I have also added a .travis.yml file so you can hook it up to automate running the tests to help with reviews.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  - "3.4"
+  - "3.5"
+  - "3.6"
+matrix:
+  include:
+    - python: 3.7
+      dist: xenial
+      sudo: true
+install: pip install -e ".[dev]"
+script: py.test --cov-report=xml --cov-report=html --cov-report term-missing --cov=awsprofile
```
Comment (jrstarke) [inline .travis.yml]: Looks like there isn't a newline at the end of the file

File: tests/test_parse_args.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+def test_parse_args_with_aws_default_profile_and_aws_profile_env_set():
+    """With AWS_PROFILE and AWS_DEFAULT_PROFILE set and no profile set on the command line it should return the correct profile name from AWS_PROFILE, the command[s] and the default cache of true """
+    argv_test_data = ['aws-profile', 'test-command-name']
+    os.environ['AWS_DEFAULT_PROFILE'] = 'profile-from-aws-default-profile-envvar'
+    os.environ['AWS_PROFILE'] = 'profile-from-aws-profile-envvar'
+    result = parse_args(argv=argv_test_data)
+    assert ('profile-from-aws-profile-envvar', ['test-command-name'], 'true') == result
+
+
+@pytest.mark.skip(reason="This test fails as the implementation in code is ambiguous is it a profile or is it part of the command? Should we add argparse and a named option?")
```
Comment (jrstarke) [inline tests/test_parse_args.py:57]: So what are you proposing? Removing the positional argument, and instead switching to `--profile <name>` or something?

File: tests/test_parse_args.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+def test_parse_args_with_aws_default_profile_env_set():
+    """With AWS_DEFAULT_PROFILE set and no profile set on the command line it should return the correct profile name from AWS_DEFAULT_PROFILE, the command[s] and the default cache of true """
+    argv_test_data = ['aws-profile', 'test-command-name']
+    os.environ['AWS_DEFAULT_PROFILE'] = 'profile-from-aws-default-profile-envvar'
+    result = parse_args(argv=argv_test_data)
+    assert ('profile-from-aws-default-profile-envvar', ['test-command-name'], 'true') == result
+
+
+@pytest.mark.skip(reason="This test fails as the implementation in code is incorrect see https://github.com/aws/aws-cli/issues/2597 which infers AWS_PROFILE should be used over AWS_DEFAULT_PROFILE")
```
Comment (jrstarke) [inline tests/test_parse_args.py:47]: So is this actually returning the `AWS_DEFAULT_PROFILE` or no value at all?

File: tests/test_parse_args.py (hunk 2 above)
Comment (grahamhar) [reply tests/test_parse_args.py:57]: I think that would be cleaner. It is just my opinion but I find positional arguments a bit vague. If merged I'll do a follow up PR to implement that.

File: tests/test_parse_args.py (hunk 3 above)
Comment (grahamhar) [reply tests/test_parse_args.py:47]: It currently returns the value of AWS_DEFAULT_PROFILE but it should return the value of  AWS_PROFILE if my understanding of issue I linked to is correct. I'll do a follow up PR to fix if you agree this should be the behaviour.

File: .travis.yml (hunk 1 above)
Comment (grahamhar) [reply .travis.yml]: Fixed

**Comment by jrstarke** [comment]
Nice Work :)

**Pull request merged**